code/hsec-tools/app/Main.hs

```diff
@@ -153,24 +155,18 @@ withAdvisory :: (Maybe FilePath -> Advisory -> IO ()) -> Maybe FilePath -> IO ()
 withAdvisory go file = do
   input <- maybe T.getContents T.readFile file
 
-  oob <- ($ emptyOutOfBandAttributes) <$> case file of
-    Nothing -> pure id
-    Just path ->
-      getAdvisoryGitInfo path <&> \case
-        Left _ -> id
-        Right gitInfo -> \oob -> oob
-          { oobPublished = Just (firstAppearanceCommitDate gitInfo)
-          , oobModified = Just (lastModificationCommitDate gitInfo)
-          }
+  oob <- runExceptT case file of
+    Nothing -> throwE StdInHasNoOOB
+    Just path -> withExceptT GitHasNoOOB do 
+      gitInfo <- ExceptT $ liftIO $ getAdvisoryGitInfo path 
```

Comment:
Isn't liftIO enough? (ExceptT should have the instance)

Reply:
`it's an `IO (Either _ _)`, `liftIO` give's you an `ExceptT IO (Either _ _)`, no?

Reply:
Sorry, I thought it was only IO